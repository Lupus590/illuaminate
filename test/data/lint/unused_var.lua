-- config: (lint (only (var:unused)))
local x, y, z, _ = 0, 0, 0
print(x)

y = 0
z[0] = 0

-- FIXME: y = 0 should not warn for unused var (unused val? Sure)
-- TODO: Is z bound or not?
