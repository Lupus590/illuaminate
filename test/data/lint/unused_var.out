unused_var.lua:[2:9-2:10]: Unused variable "y". [var:unused]
   │
 2 │ local x, y, z, _ = 0, 0, 0
   │          ^
unused_var.lua:[5:0-5:1]: Unused variable "y". [var:unused]
   │
 5 │ y = 0
   │ ^
Compilation succeeded with 2 warnings
@@ -1,8 +1,8 @@
 -- config: (lint (only (var:unused)))
-local x, y, z, _ = 0, 0, 0
+local x, _, z, _ = 0, 0, 0
 print(x)
 
-y = 0
+_ = 0
 z[0] = 0
 
 -- FIXME: y = 0 should not warn for unused var (unused val? Sure)
